src/goto-programs/goto_convert.cpp

```diff
@@ -2095,6 +2095,7 @@ symbolt &goto_convertt::new_tmp_symbol(
   const source_locationt &source_location,
   const irep_idt &mode)
 {
+  PRECONDITION(!mode.empty());
```

Comment:
Reluctant to suggest this now: but to avoid having this checked at run time - consider making mode an enum that we store in the irep using just index?